src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.OmniSharp/Project/OmniSharpTelemetryReporter.cs

```diff
@@ -12,11 +12,28 @@ public void ReportEvent(string name, Severity severity)
     {
     }
 
-    public void ReportEvent<T>(string name, Severity severity, ImmutableDictionary<string, T> values)
+    public void ReportEvent<T>(string name, Severity severity, T values) where T : IImmutableDictionary<string, object?>
```

Comment:
Why is this generic and constrained to `IImmutableDictionary<string, object?>`? Is `ImmutableDictionary<string, object?>` the only type that will ever be passed here? If so, I recommend making this method non-generic and declare it as:

```C#
void ReportEvent(string name, Severity severity, ImmutableDictionary<string, object?> values);
```